backend/internal/server/apis/schedule/pkg/optimize/optimize.go

```diff
@@ -0,0 +1,243 @@
+package optimize
+
+import (
+	"fmt"
+	"hockey/pkg/models"
+	"sort"
+)
+
+func Schedule(games []models.Game, seasonStats models.SeasonStats, teamStats map[string]models.TeamStats) {
+	resetOptimized(games)
+	fmt.Printf("OPT: Early game percent: %v%%\n", seasonStats.EarlyPercentage()*100)
+	seasonEarlyHigh := int(seasonStats.EarlyPercentage()*10.0) + 1 // TODO took a shortcut here and just hardcoded the 10 games
+	seasonEarlyLow := int(seasonStats.EarlyPercentage() * 10.0)    // - 1
+
+	// implemented as a series of constraints that have to pass in order to swap games
+
+	fmt.Printf("High count: %v\n", seasonEarlyHigh)
+	fmt.Printf("Low count: %v\n", seasonEarlyLow)
+
+	teams := make([]string, 0)
+	for key := range teamStats {
+		teams = append(teams, key)
+	}
+	sort.Strings(teams)
+
+	// Go through list of teams
+	for _, team := range teams { // Sorted teams
+		stats := teamStats[team]
+		// for team, stats := range teamStats { // Unsorted teams
+
+		// Does this team need to be balanced?
+		needsBalance, tooManyEarly := needsToBeBalanced(stats, seasonEarlyHigh, seasonEarlyLow)
+		if !needsBalance {
+			fmt.Printf("%s (%v-%v) is balanced\n", team, stats.EarlyGames, stats.LateGames)
```

Comment:
I think there is a logger in the project we should use instead of printf

see https://github.com/jak103/powerplay/blob/main/backend/internal/middleware/manager.go#L21 for example

Reply:
Yes, please update everything to the powerplay logger.